OrbitGl/TextBox.h

```diff
@@ -5,70 +5,25 @@
 #ifndef ORBIT_GL_TEXT_BOX_H_
 #define ORBIT_GL_TEXT_BOX_H_
 
-#include <cfloat>
-#include <cstdint>
-
-#include "Batcher.h"
 #include "CoreMath.h"
 #include "capture_data.pb.h"
 
 class TextRenderer;
 
 class TextBox {
  public:
-  TextBox();
-  TextBox(const Vec2& a_Pos, const Vec2& a_Size, const std::string& a_Text,
-          const Color& a_Color = Color(128, 128, 128, 128));
-
-  TextBox(const Vec2& a_Pos, const Vec2& a_Size, const Color& a_Color);
-  TextBox(const Vec2& a_Pos, const Vec2& a_Size);
-
-  void Draw(Batcher* batcher, TextRenderer& text_renderer, float min_x = -FLT_MAX,
-            bool visible = true, bool right_justify = false, bool is_inactive = false,
-            unsigned int id = 0xFFFFFFFF, bool is_picking = false, bool is_highlighted = false);
-
-  void SetSize(const Vec2& a_Size) {
-    m_Size = a_Size;
-    Update();
-  }
-  void SetSizeX(float X) {
-    m_Size[0] = X;
-    Update();
-  }
-  void SetSizeY(float Y) {
-    m_Size[1] = Y;
-    Update();
-  }
-
-  void SetPos(const Vec2& a_Pos) {
-    m_Pos = a_Pos;
-    Update();
-  }
-  void SetPosX(float X) {
-    m_Pos[0] = X;
-    Update();
-  }
-  void SetPosY(float Y) {
-    m_Pos[1] = Y;
-    Update();
-  }
-
-  const Vec2& GetSize() const { return m_Size; }
-  float GetSizeX() const { return m_Size[0]; }
-  float GetSizeY() const { return m_Size[1]; }
+  TextBox() : pos_(Vec2::Zero()), size_(Vec2(100.f, 10.f)), elapsed_time_text_length_(0){};
+  TextBox(const Vec2& pos, const Vec2& size, const std::string& text = "")
+      : pos_(pos), size_(size), text_(text), elapsed_time_text_length_(0) {}
 
-  const Vec2& GetPos() const { return m_Pos; }
-  float GetPosX() const { return m_Pos[0]; }
-  float GetPosY() const { return m_Pos[1]; }
+  void SetSize(const Vec2& size) { size_ = size; }
+  void SetPos(const Vec2& pos) { pos_ = pos; }
 
-  float GetMaxX() const { return m_Max[0]; }
-  float GetMaxY() const { return m_Max[1]; }
-
-  Vec2 GetMin() const { return m_Min; }
-  Vec2 GetMax() const { return m_Max; }
+  const Vec2& GetSize() const { return size_; }
+  const Vec2& GetPos() const { return pos_; }
 
   const std::string& GetText() const { return text_; }
-  void SetText(const std::string& a_Text) { text_ = a_Text; }
+  void SetText(std::string_view text) { text_ = text; }
```

Comment:
I think this should be a `std::string` and passed by value and then moved into the member variable. `std::string_view` would force a copy if the caller already has a string which could be moved into the member.

Reply:
Changed